shared/middleware/helpers/property_bag.rb

```diff
@@ -48,6 +51,8 @@ def set(name, value, ip_address)
   end
 
   def to_hash()
+    raise NotFound unless @can_enumerate
```

Comment:
I think it's worth considering returning and empty hash here because error conditions are harder to deal with and create multiple paths of execution. So, suggest `return {} unless @can_enumerate`.